storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -0,0 +1,518 @@
+            }
+            if (o instanceof String) {
+                return;
+            }
+            //check if iterable
+            SimpleTypeValidator isIterable = new SimpleTypeValidator();
+            try {
+                isIterable.validateField(name, Iterable.class, o);
+            } catch (Exception ex) {
+            }
+            this.fv.validateField(name, o);
+        }
+    }
+
+    /**
+     * Validates Kryo Registration
+     */
+    public static class KryoRegValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            if (o instanceof Iterable) {
+                for (Object e : (Iterable) o) {
+                    if (e instanceof Map) {
+                        for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) e).entrySet()) {
+                            if (!(entry.getKey() instanceof String) ||
+                                    !(entry.getValue() instanceof String)) {
+                                throw new IllegalArgumentException(
+                                        "Each element of the list " + name + " must be a String or a Map of Strings");
+                            }
+                        }
+                    } else if (!(e instanceof String)) {
+                        throw new IllegalArgumentException(
+                                "Each element of the list " + name + " must be a String or a Map of Strings");
+                    }
+                }
+                return;
+            }
+            throw new IllegalArgumentException(
+                    "Field " + name + " must be an Iterable containing only Strings or Maps of Strings");
+        }
+    }
+
+    /**
+     * Validates if a number is a power of 2
+     */
+    public static class PowerOf2Validator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            final long i;
+            if (o instanceof Number &&
+                    (i = ((Number) o).longValue()) == ((Number) o).doubleValue()) {
+                // Test whether the integer is a power of 2.
+                if (i > 0 && (i & (i - 1)) == 0) {
+                    return;
+                }
+            }
+            throw new IllegalArgumentException("Field " + name + " must be a power of 2.");
+        }
+    }
+
+    /**
+     * Validates each entry in a list
+     */
+    public static class ListEntryTypeValidator extends TypeValidator {
+
+        @Override
+        public void validateField(String name, Class type, Object o) {
+            ConfigValidationUtils.NestableFieldValidator validator = ConfigValidationUtils.listFv(type, false);
+            validator.validateField(name, o);
+        }
+    }
+
+    /**
+     * Validates entry in the a list with a list of custom validator
```

Comment:
Maybe this? `Validates each entry in a list against a list of custom Validators`